pkg/trace/api/api.go

```diff
@@ -143,6 +141,37 @@ func (r *HTTPReceiver) Start() {
 	}()
 }
 
+func (r *HTTPReceiver) attachDebugHandlers(mux *http.ServeMux) {
+	mux.HandleFunc("/debug/pprof/", pprof.Index)
+	mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+	mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+
+	mux.HandleFunc("/debug/blockrate", func(w http.ResponseWriter, r *http.Request) {
+		// this endpoint calls runtime.SetBlockProfileRate(v), where v is an optional
+		// query string parameter defaulting to 1.
+		rate := 1
+		v := r.URL.Query().Get("v")
+		if v != "" {
+			n, err := strconv.Atoi(v)
+			if err != nil {
+				http.Error(w, "v must be an integer", http.StatusBadRequest)
+				return
+			}
+			rate = n
+		}
+		runtime.SetBlockProfileRate(rate)
```

Comment:
I don't know. It seems nobody does. Neither the internet, nor our friends at Datadog. If you look at the standard library, it's clear that an extra code path is triggered which records blocking events. I'd say we keep it as a prevention. On top of that, it also gives us the freedom to fine-tune its value.

Reply:
Unfortunately the block profiler is [poorly documented](https://github.com/golang/go/issues/14689) and there is no mention about its impact on production environments.

All I can say is that you probably don't want/need a default value of `1`. That means that *every* blocking event will be accounted for. As far as I understand from the [documentation](https://golang.org/pkg/runtime/#SetBlockProfileRate) and the comments in the aforementioned issue, If you set it to, let's say, `100`, you will sample one blocking event for every `100ns` spent blocked, which I assume will have a lower overhead and will still be useful enough to track things like mutexes with high contention.

Reply:
In that case it might be reasonable to have it enabled at all times with a higher default (say 1000 or even much more)

Reply:
Ok, so I've done further digging around, and discovered that [this function](https://github.com/golang/go/blob/master/src/runtime/mprof.go#L414-L428) gets called every time a block profile entry is recorded. That function in turn generates a traceback every time, which we know from previous findings that it can be super costly. Quite [a lot of things](https://github.com/golang/go/blob/master/src/runtime/traceback.go#L98) are happening there.

For this reason, I updated the PR to default to the value 10000 (1 sample every 10 microseconds) and I'll suggest keeping this endpoint and not having it on by default.